pkg/common/scheduler/README.md

```diff
@@ -0,0 +1,247 @@
+
+#### 6. Evictor
+
+Autoscaler periodically attempts to compact veplicas into a smaller number of free replicas with lower ordinals. Vreplicas placed on higher ordinal pods are evicted and rescheduled to pods with a lower ordinal using the same scheduling strategies.
+
+### Scheduler Profile
+
+#### Predicates:
+
+1. **PodFitsResources**: check if a pod has enough capacity [CORE]
+
+2. **NoMaxResourceCount**: check if total number of placement pods exceed available resources  [KAFKA]. It has an argument `NumPartitions` to configure the plugin with the total number of Kafka partitions.
+
+3. **EvenPodSpread**: check if resources are evenly spread across pods [CORE]. It has an argument `MaxSkew` to configure the plugin with an allowed skew factor.
+
+#### Priorities:
+
+1. **AvailabilityNodePriority**: make sure resources are evenly spread across nodes [CORE]. It has an argument `MaxSkew` to configure the plugin with an allowed skew factor.
+
+2. **AvailabilityZonePriority**: make sure resources are evenly spread across zones [CORE]. It has an argument `MaxSkew` to configure the plugin with an allowed skew factor.
+
+3. **LowestOrdinalPriority**: make sure vreplicas are placed on free smaller ordinal pods to minimize resource usage [CORE]
+
+**Example profile:**
+
+```
+data:
+  predicates: |+
+                  [
+                    {"Name": "PodFitsResources"},
+                    {"Name": "NoMaxResourceCount",
+                    "Args": "{\"NumPartitions\": 100}"},
+                    {"Name": "EvenPodSpread",
+                    "Args": "{\"MaxSkew\": 2}"}
+                  ]
+  priorities: |+
+                  [
+                    {"Name": "AvailabilityZonePriority",
+                    "Weight": 10,
+                    "Args":  "{\"MaxSkew\": 2}"},
+                    {"Name": "LowestOrdinalPriority",
+                    "Weight": 2}
+                  ]
+```
+
+### Descheduler Profile:
+
+#### Priorities:
+
+1. **RemoveWithAvailabilityNodePriority**: make sure resources are evenly spread across nodes [CORE]
+
+2. **RemoveWithAvailabilityZonePriority**: make sure resources are evenly spread across zones [CORE]
+
+3. **HighestOrdinalPriority**: make sure vreps are removed from higher ordinal pods to minimize resource usage [CORE]
+
+**Example profile:**
+
+```
+data:
+  priorities: |+
+                  [
+                    {"Name": "RemoveWithEvenPodSpreadPriority",
+                    "Weight": 10,
+                    "Args": "{\"MaxSkew\": 2}"},
+                    {"Name": "RemoveWithAvailabilityZonePriority",
+                    "Weight": 10,
+                    "Args":  "{\"MaxSkew\": 2}"},
+                    {"Name": "RemoveWithHighestOrdinalPriority",
+                    "Weight": 2}
+                  ]
+```
+
+### Recovery
```

Comment:
In addition to recovery scenario, there should be a section about normal operation, such as Knative upgrade and K8s upgrade. I wouldn't do it in this PR, just something to keep in mind in the future.

Reply:
Adding a preliminary section for this here. More details as we use KPerf can come in the future.